src/test/java/uk/gov/hmcts/juror/api/moj/report/grouped/UnconfirmedAttendanceReportTest.java

```diff
@@ -0,0 +1,160 @@
+            DataType.APPEARANCE_CHECKED_OUT);
+
+        setHasPoolRepository(false);
+    }
+
+    @BeforeEach
+    @Override
+    public void beforeEach() {
+        securityUtilMockedStatic = mockStatic(SecurityUtil.class);
+        courtLocationService = mock(CourtLocationService.class);
+        super.beforeEach();
+    }
+
+    @AfterEach
+    void afterEach() {
+        securityUtilMockedStatic.close();
+    }
+
+    @Override
+    public UnconfirmedAttendanceReport createReport(PoolRequestRepository poolRequestRepository) {
+        return new UnconfirmedAttendanceReport(this.courtLocationService);
+    }
+
+    @Override
+    protected StandardReportRequest getValidRequest() {
+        return StandardReportRequest.builder()
+            .reportType("UnconfirmedAttendanceReport")
+            .fromDate(LocalDate.of(2024, 1, 1))
+            .toDate(LocalDate.of(2024, 1, 2))
+            .build();
+    }
+
+    @Override
+    public void positivePreProcessQueryTypical(JPAQuery<Tuple> query, StandardReportRequest request) {
+        request.setFromDate(LocalDate.of(2024, 1, 1));
+        request.setToDate(LocalDate.of(2024, 1, 2));
+        securityUtilMockedStatic.when(SecurityUtil::getLocCode).thenReturn(TestConstants.VALID_COURT_LOCATION);
+        report.preProcessQuery(query, request);
+
+        verify(query).where(QAppearance.appearance.attendanceDate.between(request.getFromDate(), request.getToDate())
+            .and(QAppearance.appearance.appearanceStage.in(
+                AppearanceStage.CHECKED_IN,
+                AppearanceStage.CHECKED_OUT
+            )));
+        verify(query).where(QAppearance.appearance.locCode.eq(SecurityUtil.getLocCode()));
+        verify(query).orderBy(QAppearance.appearance.attendanceDate.desc(), QAppearance.appearance.jurorNumber.asc());
+    }
+
+    @Override
+    public Map<String, AbstractReportResponse.DataTypeValue> positiveGetHeadingsTypical(
+        StandardReportRequest request,
+        AbstractReportResponse.TableData<GroupedTableData> tableData,
+        GroupedTableData data) {
+
+        request.setFromDate(LocalDate.of(2024, 1, 1));
+        request.setToDate(LocalDate.of(2024, 1, 2));
+        securityUtilMockedStatic.when(SecurityUtil::getLocCode).thenReturn(TestConstants.VALID_COURT_LOCATION);
+
+        when(data.getSize()).thenReturn(50);
+
+        CourtLocation courtLocation = mock(CourtLocation.class);
+        when(courtLocationService.getCourtLocation(TestConstants.VALID_COURT_LOCATION)).thenReturn(courtLocation);
+        doReturn(getCourtNameEntry()).when(report).getCourtNameHeader(courtLocation);
+
+        Map<String, GroupedReportResponse.DataTypeValue> expected = new ConcurrentHashMap<>();
+        expected.put("total_unconfirmed_attendances", GroupedReportResponse.DataTypeValue.builder()
+             .displayName("Total unconfirmed attendances")
+             .dataType(Long.class.getSimpleName())
+             .value(tableData.getData().getSize())
```

Comment:
This value should be hard coded to ensure it is what you are expecting

Reply:
updated